code/TestCategory.php

```diff
@@ -36,7 +36,7 @@ public static function map()
 
     public function requireDefaultRecords()
     {
-        if (!DataObject::get_one(static::class)) {
+        if (static::get()->count() === 0) {
```

Comment:
No need for caching here - what we really want to know is "are there any records here?"